### ISSUE TITLE
ncdu listing failed: strconv.ParseInt: parsing "1.54854926086E 12": invalid syntax

### ISSUE THREAD

**Issue by geoffmyers** [ncw/rclone#3065]
#### What is the problem you are having with rclone?

I am trying to run `rclone ncdu` on a Backblaze B2 bucket. It starts listing files and directories inside of the bucket but then after about 5 seconds, it consistently fails with the following error message:

```
Failed to ncdu: ncdu directory listing: ncdu listing failed: strconv.ParseInt: parsing "1.54854926086E 12": invalid syntax
```

#### What is your rclone version (output from `rclone version`)

```
rclone v1.46
- os/arch: darwin/amd64
- go version: go1.11.5
```

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)

Apple macOS Mojave 10.14.3 (18D109) [64-bit]

####  Which cloud storage system are you using? (eg Google Drive)

Backblaze B2

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)

```
rclone ncdu b2:<bucket>
```

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)

```
2019/03/22 16:39:57 DEBUG : rclone: Version "v1.46" starting with parameters ["rclone" "-vv" "ncdu" "b2:<bucket>"]
2019/03/22 16:39:57 DEBUG : Using config file from "~/.config/rclone/rclone.conf"
2019/03/22 16:40:03 Failed to ncdu: ncdu directory listing: ncdu listing failed: strconv.ParseInt: parsing "1.54854926086E 12": invalid syntax
```

**Comment by ncw** [comment]
> strconv.ParseInt: parsing "1.54854926086E 12": invalid syntax

What that means is that the JSON info backblaze is sending back has a floating point number in it where rclone is expecting an integer.  That particular floating point number looks malformed too which makes me wonder if this is a b2 bug...

1.5E12 would be 1.5 TB - do you have any files that big? If you do then just running `rclone ls b2:bucket/path/to/file` will probably provoke the bug.

I'd really like to see the request and response from b2.  You can get this by adding `-vv --log-file b2.log --dump bodies` to your `rclone ncdu` command.

Can you attach (or email me at nick@craig-wood.com with a link to this issue) that log?  Or maybe you can figure out which one the bad response is and just post that and the corresponding request (you can match them up with the hex number).

If it is a b2 bug I know the right people to report it to :-)

**Comment by geoffmyers** [comment]
Thanks for your help! Yes, I do believe this is a Backblaze B2 bug because, as you said, the JSON response appears to have a malformed value for `src_last_modified_millis`. Here is the relevant excerpt of the debug info you requested:

```
2019/03/25 11:00:14 DEBUG : >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
2019/03/25 11:00:14 DEBUG : <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
2019/03/25 11:00:14 DEBUG : HTTP RESPONSE (req 0xc0007e6200)
2019/03/25 11:00:14 DEBUG : HTTP/1.1 200 
Content-Length: 674
Cache-Control: max-age=0, no-cache, no-store
Content-Type: application/json;charset=utf-8
Date: Mon, 25 Mar 2019 15:00:14 GMT

{
  "files": [
    {
      "accountId": "2de109e72d4d",
      "action": "upload",
      "bucketId": "123d4ed1b039ae7752fd041d",
      "contentLength": 2036142,
      "contentSha1": "884c14600c932f08c002f3a28a6716e22429c0b6",
      "contentType": "application/octet-stream",
      "fileId": "4_z123d4ed1b039ae7752fd041d_f1130e454287f32b6_d20190325_m000809_c000_v0001056_t0015",
      "fileInfo": {
        "src_last_modified_millis": "1.55347248875E 12"
      },
      "fileName": "[redacted]",
      "size": 2036142,
      "uploadTimestamp": 1553472489000
    }
  ],
  "nextFileName": null
}
2019/03/25 11:00:14 DEBUG : <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
2019/03/25 11:00:14 DEBUG : [redacted]: Failed to parse mod time string "1.55347248875E 12": strconv.ParseInt: parsing "1.55347248875E 12": invalid syntax
2019/03/25 11:00:14 ERROR : [redacted]: error listing: strconv.ParseInt: parsing "1.55347248875E 12": invalid syntax
2019/03/25 11:00:14 DEBUG : >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
2019/03/25 11:00:14 DEBUG : HTTP REQUEST (req 0xc0007e6600)
2019/03/25 11:00:14 DEBUG : POST /b2api/v1/b2_list_file_names HTTP/1.1
Host: api000.backblazeb2.com
User-Agent: rclone/v1.46
Content-Length: 150
Authorization: XXXX
Content-Type: application/json
Accept-Encoding: gzip
```

**Comment by ncw** [comment]
> Thanks for your help! Yes, I do believe this is a Backblaze B2 bug because, as you said, the JSON response appears to have a malformed value for `src_last_modified_millis` (i.e. space character between "E" and "12"). Here is the relevant excerpt of the debug info you requested:

Ah, that is insteresting!

The "fileInfo" object contains things created by the clients, not backblaze though...

What program uploaded this file? I think that is where the bug is!

I could make rclone just carry on if it can't parse the modified time.

**Comment by ncw** [comment]
I've changed rclone so it will emit that debug message but carry on if the `src_last_modified_millis` is malformed.

Hopefully that will workaround your problem - can you give it a go?

https://beta.rclone.org/branch/v1.46.0-109-g6fb1c8f5-fix-3065-b2-time-parse-beta/ (uploaded in 15-30 mins)

**Comment by geoffmyers** [comment]
I can confirm that the beta version of rclone resolves the issue (ignores the malformed value) and does not cause commands such as `ncdu` or `ls` to crash. Thanks so much!

**Comment by ncw** [comment]
Thanks for testing!

I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.47
